warehouse/query-core/src/test/java/datawave/query/jexl/visitors/FieldMissingFromDateRangeVisitorTest.java

```diff
@@ -0,0 +1,307 @@
+package datawave.query.jexl.visitors;
+
+import static datawave.data.ColumnFamilyConstants.COLF_F;
+import static datawave.query.Constants.ANY_FIELD;
+import static datawave.query.Constants.NO_FIELD;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import org.apache.accumulo.core.client.AccumuloClient;
+import org.apache.accumulo.core.client.BatchWriter;
+import org.apache.accumulo.core.client.BatchWriterConfig;
+import org.apache.accumulo.core.client.MutationsRejectedException;
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.LongCombiner;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.commons.jexl3.parser.ASTJexlScript;
+import org.apache.commons.jexl3.parser.ParseException;
+import org.apache.hadoop.io.Text;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+import datawave.accumulo.inmemory.InMemoryAccumuloClient;
+import datawave.accumulo.inmemory.InMemoryInstance;
+import datawave.microservice.query.Query;
+import datawave.query.composite.CompositeMetadataHelper;
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.util.AllFieldMetadataHelper;
+import datawave.query.util.MetadataHelper;
+import datawave.query.util.TypeMetadataHelper;
+
+public class FieldMissingFromDateRangeVisitorTest {
+    private MetadataHelper helper;
+    private Query querySettings;
+    private AccumuloClient accumuloClient;
+
+    // Special fields required by visitor.
+    private final Set<String> specialFields = Sets.newHashSet(ANY_FIELD, NO_FIELD);
+    private final List<Mutation> mutations = new ArrayList<>();
+
+    private static final String TABLE_METADATA = "metadata";
+    private static final String[] AUTHS = {"FOO", "BAR"};
+    private static final Set<Authorizations> AUTHS_SET = Collections.singleton(new Authorizations(AUTHS));
+    private static final String NULL_BYTE = "\0";
+
+    @BeforeEach
+    public void setUp() throws Exception {
```

Comment:
You can avoid the overhead of using the InMemoryAccumuloClient here by either passing in a mocked MetadataHelper or an instance of datawave.query.util.MockMetadataHelper. That way your tests do not care about how MetadatHelper is implemented.